invenio_records_resources/records/models.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# Invenio-Records-Resources is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Records Models."""
+
+import invenio_records.models as base_models
+from invenio_db import db
+from invenio_files_rest.models import Bucket, ObjectVersion
+from sqlalchemy.dialects import mysql
+from sqlalchemy.ext.declarative import declared_attr
+from sqlalchemy_utils.types import UUIDType
+
+
+class RecordFileBase(base_models.RecordMetadataBase):
+    """Base class for a record file, storing its state and metadata."""
+
+    record_model_cls = None
+    """."""
+
+    key = db.Column(
+        db.Text().with_variant(mysql.VARCHAR(255), 'mysql'),
+        nullable=False,
+    )
+    """Filename key (can be path-like also)."""
+
+    @declared_attr
+    def record_id(cls):
+        """Record ID foreign key."""
+        return db.Column(
+            UUIDType,
+            db.ForeignKey(cls.record_model_cls.id, ondelete='RESTRICT'),
+            nullable=False,
+        )
+
+    @declared_attr
+    def record(cls):
+        """Record the file belnogs to."""
+        return db.relationship(cls.record_model_cls)
+
+    @declared_attr
+    def object_version_id(cls):
+        """Object version ID foreign key."""
+        return db.Column(
+            UUIDType,
+            db.ForeignKey(ObjectVersion.version_id, ondelete='RESTRICT'),
+            nullable=True,
+        )
+
+    @declared_attr
+    def object_version(cls):
+        """Object version connected to the record file."""
+        return db.relationship(ObjectVersion)
+
+    @declared_attr
+    def __table_args__(cls):
+        """Table args."""
+        return (
+            db.Index(
+                f'uidx_{cls.__tablename__}_id_key',
+                'id', 'key', unique=True,
+            ),
+        )
+
+
+class RecordMetadataBase(base_models.RecordMetadataBase):
+    """Represent a base class for record metadata with files."""
+
+    @declared_attr
+    def bucket_id(cls):
```

Comment:
comment: if we add a `bucket` column on the db level, we are assuming there is no chance of using records without files (e.g. someone implementing their app using only metadata). Or at least they will have an empty column in every row. This is fine for me, even more taking into account the time constraint. However, given this, what we discussed at service level could be made tight with files (still with mixins) but without taking into account the no files case.

Reply:
I agree. I've removed it from the default classes and added it only in tests. This will be added concretely in `invenio-rdm-records`.